bin/genesis

```diff
@@ -4445,6 +4452,49 @@ EOF
 		echo
 		bosh target ${target}
 		bosh deployment ${DEPLOYMENT_ENV_DIR}/manifests/.deploy.yml
+
+		if [[ ${CI_WAIT_FOR_CLEAN_BOSH} == "true" ]]; then
+			i=1
+			attempts=10
+			wait=10
+			echo "Checking if BOSH has any tasks running... (for up to $(( $attempts * $wait )) seconds)"
+
+			while (( $i <= $attempts ))
+			do
+				tasks=$(ask_bosh '/tasks?state=queued,processing,cancelling&verbose=2')
```

Comment:
This asks the wrong bosh for tasks. It asks the bosh which will deploy the current deployment, rather than the bosh of the current deployment.